src/main/java/com/synopsys/integration/jira/common/cloud/service/IssueService.java

```diff
@@ -67,30 +70,80 @@ public IssueCreationResponseModel createIssue(IssueCreationRequestModel requestM
         String projectName = requestModel.getProjectName();
         String reporterEmail = requestModel.getReporterEmail();
 
-        IssueTypeResponseModel foundIssueType = issueTypeService.getAllIssueTypes().stream()
-                                                    .filter(issueType -> issueType.getName().equalsIgnoreCase(issueTypeName))
-                                                    .findFirst()
-                                                    .orElseThrow(() -> new JiraPreconditionNotMetException(String.format("Issue type not found; issue type %s", issueTypeName)));
         PageOfProjectsResponseModel pageOfProjects = projectService.getProjectsByName(projectName);
-        ProjectComponent foundProject = pageOfProjects.getProjects().stream()
-                                            .findFirst()
-                                            .orElseThrow(() -> new JiraPreconditionNotMetException(String.format("Project not found; project name: %s", projectName)));
+        String projectId = pageOfProjects.getProjects()
+                               .stream()
+                               .findFirst()
+                               .map(ProjectComponent::getId)
+                               .orElseThrow(() -> new JiraPreconditionNotMetException(String.format("Project not found; project name: %s", projectName)));
+
+        List<IssueTypeResponseModel> allIssueTypes = issueTypeService.getAllIssueTypes();
+        List<IssueTypeResponseModel> issueTypeCandidates = allIssueTypes
+                                                               .stream()
+                                                               .filter(issueType -> issueType.getName().equalsIgnoreCase(issueTypeName))
+                                                               .collect(Collectors.toList());
+        String issueTypeId = null;
+        for (IssueTypeResponseModel issueTypeCandidate : issueTypeCandidates) {
+            IssueTypeScope scope = issueTypeCandidate.getScope();
+            if (null != scope) {
+                IdComponent scopedProject = scope.getProject();
+                if (null != scopedProject) {
+                    issueTypeId = scopedProject.getId();
+                    // Exact match. Search complete.
+                    break;
+                }
+            } else {
+                // Possible match. There could be a better one.
+                issueTypeId = issueTypeCandidate.getId();
+            }
```

Comment:
Could this not be done in the stream when going through the issue types?

Reply:
It could, but it would be significantly less readable.